File: server/FitMate.Core/JsonModels/Subscriptions/EffectiveEntitlementsModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Subscriptions;

public class EffectiveEntitlementsModel
{
    public long PlanId { get; set; }
    public string PlanCode { get; set; } = string.Empty;
    public string PlanName { get; set; } = string.Empty;
    public EntitlementSource Source { get; set; }
    public List<FeatureAvailabilityModel> Features { get; set; } = [];
}

File: client/src/types/JsonModels/Subscriptions/EffectiveEntitlementsModel.ts
import type { JsonModels } from "../../backend";
import type { EntitlementSource } from "../Enums/EntitlementSource";

export interface EffectiveEntitlementsModel
{
	planId: number;
	planCode: string;
	planName: string;
	source: EntitlementSource;
	features: JsonModels.Subscriptions.FeatureAvailabilityModel[];
}
